pkg/api/util/ipc_endpoint.go

```diff
@@ -0,0 +1,88 @@
+package util
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/system"
+)
+
+type IPCEndpoint struct {
+	client    *http.Client
+	target    url.URL
+	closeConn bool
+}
+
+func (end *IPCEndpoint) SetCloseConnection(state bool) {
+	end.closeConn = state
+}
+
+// send GET method to the endpoint
+func (end *IPCEndpoint) DoGet() ([]byte, error) {
```

Comment:
We would benefit from adding unit tests for this new utility struct and functionality 😄

Reply:
Good call, done.